rmw_connext_dynamic_cpp/src/functions.cpp

```diff
@@ -188,165 +225,17 @@ DDS_TypeCode * create_type_code(
-          if (!member_type_code) {
-            // error string was set within the function
-            goto fail;
-          }
-        }
-        break;
-      default:
-        RMW_SET_ERROR_MSG(
-          (std::string("unknown type id ") + std::to_string(member->type_id_)).c_str());
-        goto fail;
-    }
-    if (!member_type_code) {
-      RMW_SET_ERROR_MSG("failed to create typecode");
-      goto fail;
-    }
-    if (member->is_array_) {
-      if (member->array_size_) {
-        if (member->array_size_ > (std::numeric_limits<DDS_UnsignedLong>::max)()) {
-          RMW_SET_ERROR_MSG("failed to create array typecode since the size exceeds the DDS type");
-          goto fail;
-        }
-        DDS_UnsignedLong array_size = static_cast<DDS_UnsignedLong>(member->array_size_);
-        if (!member->is_upper_bound_) {
-          member_type_code_non_const = factory->create_array_tc(array_size, member_type_code, ex);
-          member_type_code = member_type_code_non_const;
-          if (!member_type_code || ex != DDS_NO_EXCEPTION_CODE) {
-            RMW_SET_ERROR_MSG("failed to create array typecode");
-            goto fail;
-          }
-        } else {
-          member_type_code_non_const =
-            factory->create_sequence_tc(array_size, member_type_code, ex);
-          member_type_code = member_type_code_non_const;
-          if (!member_type_code || ex != DDS_NO_EXCEPTION_CODE) {
-            RMW_SET_ERROR_MSG("failed to create sequence typecode");
-            goto fail;
-          }
-        }
-      } else {
-        member_type_code_non_const = factory->create_sequence_tc(
-          RTI_INT32_MAX, member_type_code, ex);
-        member_type_code = member_type_code_non_const;
-        if (!member_type_code || ex != DDS_NO_EXCEPTION_CODE) {
-          RMW_SET_ERROR_MSG("failed to create sequence typecode");
-          goto fail;
-        }
-      }
-    }
-    auto zero_based_index = type_code->add_member(
-      (std::string(member->name_) + "_").c_str(),
-      i,
-      member_type_code,
-      DDS_TYPECODE_NONKEY_REQUIRED_MEMBER, ex);
-    if (ex != DDS_NO_EXCEPTION_CODE) {
-      RMW_SET_ERROR_MSG("failed to add member");
-      goto fail;
-    }
-    if (zero_based_index != i) {
-      RMW_SET_ERROR_MSG("unexpected member index");
-      return NULL;
-    }
-  }
-  // since empty message definitions are not supported
-  // we have to add the same dummy field as in rosidl_generator_dds_idl
-  if (members->member_count_ == 0) {
-    const DDS_TypeCode * member_type_code;
-    member_type_code = factory->get_primitive_tc(DDS_TK_BOOLEAN);
-    if (!member_type_code) {
-      RMW_SET_ERROR_MSG("failed to get primitive typecode");
-      goto fail;
-    }
-    type_code->add_member("_dummy", DDS_TYPECODE_MEMBER_ID_INVALID,
-      member_type_code,
-      DDS_TYPECODE_NONKEY_REQUIRED_MEMBER, ex);
-    if (ex != DDS_NO_EXCEPTION_CODE) {
-      RMW_SET_ERROR_MSG("failed to add member");
-      goto fail;
-    }
```

Comment:
Since moving such big blocks of code makes it very difficult to review code changes I would suggest to have a separate commit (which doesn't get squashed) which only moves the code from the one location to the new location (without changing it). Then this specific commit can be ignored after a short glance.